compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -829,6 +829,18 @@ impl<'a> Parser<'a> {
         if let Some(lt) = lifetime {
             self.error_remove_borrow_lifetime(span, lt.ident.span.until(expr.span));
         }
+
+        // Add expected tokens if we parsed `&raw` as an expression.
+        // This will make sure we see "expected `const`, `mut`", and
+        // guides recovery in case we write `&raw expr`.
+        if borrow_kind == ast::BorrowKind::Ref
+            && mutbl == ast::Mutability::Not
+            && matches!(&expr.kind, ExprKind::Path(None, p) if p.is_ident(kw::Raw))
```

Comment:
(minor) `is_ident` or this `matches!` needs to account for raw identifiers, so we can exclude `r#raw` (*raw raw*) here

Reply:
This is a parsed ident, not an ident token, so we don't have a way to distinguish `r#raw` here, since ident is just a span and a symbol.

Reply:
Hmm 🤔 why is this being checked on the parsed expr and not as a lookahead after parsing the borrow modifiers? I guess kinda related to the recovery comment above, but would trivially allow checking for non-raw ident

Reply:
Well we do want to avoid putting the expectation down for something like `&raw.field`